### ISSUE TITLE
customTypeMappings side effects

### ISSUE THREAD

**Issue by anortham** [jburzynski/TypeGen#67]
I want to set string types to default to a union type of string | null. 

````
"customTypeMappings": {
    "System.DateTime": "string | null",
    "System.String":  "string | null" 
  }
````

This has the expected result for string types but not for string arrays:

```
export interface SecurityRole {
  roleId: string;
  name: string | null;
  description: string | null;
  permissions: string | null[];
}
```

**Comment by jburzynski** [comment]
Thanks for posting, I'll look into this and post any updates here.

**Comment by jburzynski** [comment]
I just released 2.4.2 in which there is an option to specify type unions for a member with `TsTypeUnionsAttribute` or in the generation spec.

I also added the `typeUnionsForTypes` config parameter which allows to specify type unions per TypeScript type.

The `typeUnionsForTypes` parameter is not essential if you're using generation specs (because you can iterate through type's members and add `.TsTypeUnions()` for all members of a given type/shape), but it may make things easier in some cases.

**Comment by anortham** [comment]
Thanks. This is looks great.
